src/Chainweb/Miner/POW.hs

```diff
@@ -132,120 +156,245 @@ powMiner logFun conf nid cutDb = runForever logFun "POW Miner" $ do
         --
         go gen (i + 1) (HM.filter (\(T2 h _) -> h > limit) adjustments')
 
-    -- | INVARIANT: A new cut, chain, and parent header is reselected after each
-    -- hash failure. This ensures that the Cut the miner is working on doesn't
-    -- grow stale, and cause forks. Without this condition (or a similar one),
-    -- the miners cause forks quite aggressively.
+awaitNextCut :: CutDb cas -> Cut -> IO Cut
+awaitNextCut cutDb c = atomically $ do
+    c' <- _cutStm cutDb
+    when (c' == c) retry
+    return c'
```

Comment:
So, this allows the Miner to continue mining only so long as the `Cut` he has is the newest one. Instead of polling for a new one all the time like before, this approach is "event" based. Okay, cool.